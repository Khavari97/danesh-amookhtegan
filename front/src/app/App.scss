@import 'assets/fonts/fonts';

* {
  margin: 0;
  font-family: "Shabnam", serif;
}

a {
  text-decoration: none;
}

//Slider css
.alice-carousel__dots {
  .alice-carousel__dots-item:not(.__custom).__active {
    background-color: #046B00;
  }
}


