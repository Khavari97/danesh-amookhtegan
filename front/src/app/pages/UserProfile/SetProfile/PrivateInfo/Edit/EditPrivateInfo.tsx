import React, {Component} from 'react';
import Style from './../../SetProfile.module.scss'
import {Button} from "../../../../../components/material-ui"
import {Formik, FormikProps, FormikValues} from "formik";
import CustomInputs from "../../../../../components/customs/input/input";
import {
    Dialog,
    DialogActions,
    DialogContent,
    DialogContentText,
    DialogTitle,
    Grid,
    IconButton,
    ListItem
} from '@material-ui/core';
import images from './../../../../../assets/images/userProfile'
import Select from './../../../../../components/customs/select/select';
import {connect} from 'react-redux'
import {GetProfile, ProvinceApi} from "../../../../../redux/User/Profile/Actions";
import {rootState} from "../../../../../redux/RootReducer";
import {apiClient} from './../../../../../api/_api'
import * as yup from "yup";
import Snackbar from "./../../../../../components/material-ui/Snackbar/SnackbarUtils";

interface IState {
    fields?: object | any;
    emailModal: boolean;
    initial: object | any

}

interface IProps {
    forwardedRef: any
    provinces: any
    cities: any
    GetProfile: typeof GetProfile
    universities: any
    data: any
    description: string
    postData?:any
}

const mapStateToProps = (state: rootState) => {
    return {
        provinces: state.userProfile.provinces,
        universities: state.userProfile.universities,
        cities: state.userProfile.cities,
        data: state.userProfile.data,
        description: state.userProfile.description
    }
}

const mapDispatchToProps = (dispatch: any) => {
    return {
        ProvinceApi: () => dispatch(ProvinceApi()),
        GetProfile: () => dispatch(GetProfile())
    }
}


class PrivateInfo extends Component<IProps, IState> {
    private formRef: any;

    constructor(props: IProps) {
        super(props);
        this.formRef = React.createRef()
        this.state = {
            fields: [
                [{
                    id: 'signInputs',
                    type: "text",
                    name: "firstName",
                    label: "نام",
                    placeholder: "",
                },
                    {
                        id: 'signInputs',
                        type: "text",
                        name: "lastName",
                        label: "نام خانوادگی",
                        placeholder: "",
                    },
                ],
                {
                    id: 'signInputs',
                    type: "text",
                    name: "gender",
                    label: "جنسیت",
                    items: [{name: "مرد", id: "MALE"}, {name: "زن", id: "FEMALE"}],
                    placeholder: "",
                },
                {
                    id: 'signInputs',
                    type: "text",
                    name: "username",
                    label: "نام کاربری",
                    placeholder: "",
                },
                {
                    id: 'signInputs',
                    type: "email",
                    name: "email",
                    label: "ایمیل",
                    placeholder: "",
                    value: ''
                },
                [
                    {
                        id: 'signInputs',
                        type: "text",
                        name: "province",
                        label: "استان",
                        placeholder: "",
                        items: this.props.provinces,
                    },
                    {
                        id: 'signInputs',
                        type: "text",
                        name: "city",
                        label: "شهر",
                        placeholder: "",
                        items: this.props.cities,
                    },
                ],
                {
                    id: 'signInputs',
                    type: "text",
                    name: "university",
                    label: "دانشگاه",
                    placeholder: "",
                    items: this.props.universities,
                },
                {
                    id: 'signInputs',
                    type: "text",
                    name: "phoneNumber",
                    label: "شماره تلفن",
                    placeholder: "",
                    value: '+98'
                },
            ],
            emailModal: false,
            initial: {
                firstName: this.props.data?.firstName,
                lastName: this.props.data?.lastName,
                email: this.props.data?.email,
                username: this.props?.data.username,
                province: this.props?.data.provinceObject,
                city: this.props.data?.cityObject,
                university: this.props.data?.universityObject,
                phoneNumber: (this.props.data?.phoneNumber == null || this.props.data.phoneNumber == '')
                    ? '+98' : this.props.data?.phoneNumber,
                description: this.props?.description,
                gender: this.props.data?.gender
            }
        }
    }

    componentDidUpdate() {
        if (this.state.fields[4][1].items != undefined) {
            if (!this.arraysAreEqual(this.props.cities, this.state.fields[4][1].items)) // Check if it's a new user, you can also use some unique property, like the ID  (this.props.user.id !== prevProps.user.id)
            {
                let fields = [...this.state.fields]
                fields[4][1].items = this.props.cities
                this.setState({fields: fields})
            }
        }
    }

    arraysAreEqual = (ary1: any, ary2: any) => {
        return (ary1.join('') == ary2.join(''));
    }
    handleModalOpen = () => {
        this.setState({emailModal: !this.state.emailModal})
    }

    emailDialog = () => {
        return (
            <div>
                <IconButton aria-label="delete" onClick={this.handleModalOpen}>
                    <img style={{width: '20px'}} alt="" src={images.alert}/>
                </IconButton>
                <Dialog
                    open={this.state.emailModal}
                    onClose={this.handleModalOpen}
                    aria-labelledby="alert-dialog-title"
                    aria-describedby="alert-dialog-description"
                >
                    <DialogTitle id="alert-dialog-title">{" توضیحات"}</DialogTitle>
                    <DialogContent>
                        <DialogContentText id="alert-dialog-description">
                            {" ایمیل شما به دلیل تایید و ذخیره در سایت غیر قابل تغییر است"}
                        </DialogContentText>
                    </DialogContent>
                    <DialogActions>
                        <Button variant="contained" onClick={this.handleModalOpen} color="primary">
                            فهمیدم
                        </Button>
                    </DialogActions>
                </Dialog>
            </div>
        )
    }


    selectItems = (element: any, props: FormikProps<any>) => {
        return (
            <Select object={{
                ...element,
                data_test:element.name,
                value: props.values[element.name],
                setFieldValue: props.setFieldValue,
                onChange: props.handleChange
            }}></Select>
        )
    }


    inputs = (element: any, index: string, props: FormikProps<any>) => {
        if (element instanceof Array) {
            return element.map((item: any, index2: number) => {
                if (item.items !== undefined) {
                    return (
                        <Grid item className={Style.gridInputs} xs={12} md={5} key={index + index2}>
                            {this.selectItems(item, props)}
                        </Grid>
                    )

                } else {
                    return (
                        <Grid item className={Style.gridInputs} xs={12} md={5} key={index + index2}>
                            <CustomInputs items={{
                                ...item,
                                data_test:item.name,
                                error: Boolean(props.errors[item.name]),
                                helperText: (props.errors[item.name]),
                                value: props.values[item.name],
                                onChange: props.handleChange
                            }}/>
                        </Grid>
                    )
                }


            })
        } else {
            if (element.name === 'email') {
                return (
                    <React.Fragment key={index}>
                        <div key={index} data-testid={element.name}>
                            <ListItem><p className={Style.title}>{element.label}</p></ListItem>
                            <ListItem><p className={Style.description}>{this.props.data.email}</p></ListItem>
                        </div>
                        {this.emailDialog()}
                    </React.Fragment>
                )
            }
            if (element.items !== undefined) {
                return (
                    <Grid item className={Style.gridInputs} xs={12} md={5} key={index}>
                        {this.selectItems(element, props)}
                    </Grid>
                )

            } else {
                return (
                    <Grid xs={12} md={5} className={Style.gridInputs} item key={index}>
                        <CustomInputs items={{
                            ...element,
                            data_test:element.name,
                            error: Boolean(props.errors[element.name]),
                            helperText: (props.errors[element.name]),
                            value: props.values[element.name],
                            onChange: props.handleChange
                        }}/>
                    </Grid>
                )
            }


        }
    }

    items = (props: FormikProps<any>) => {
        return this.state.fields.map((element: any, index: number) => {
            return (
                <Grid key={index} style={{display: 'flex'}} container className={Style.gridItems} item xs={12} sm={6}>
                    {this.inputs(element, 'input' + index, props)}
                </Grid>

            )
        })
    }

    postData = (values: FormikValues) => {
        let data = {
            firstName: values.firstName,
            lastName: values.lastName,
            username: values.username,
            email: values.email,
            university: (values.university) ? values.university : null,
            city: (values.city) ? values.city : null,
            phoneNumber: (values.phoneNumber == '') ? null
                : (values.phoneNumber.startsWith('+98')) ? ((values.phoneNumber.substring(3) == '') ? null
                    : values.phoneNumber) :
                    '+98' + ((values.phoneNumber.startsWith('0')) ? values.phoneNumber :
                    values.phoneNumber.substring(0)),
            description: this.props.description,
            gender: (values.gender) ? values.gender : null,
        }
        apiClient.patch('user/profile/', {data: data})
            .then(() => {
                this.props.GetProfile()
            })
            .catch((err) => {
                console.log(err)
                if (err.status == 400) {
                    for (const [key, value] of Object.entries(err.data.detail)) {
                        Snackbar.error('' + value)
                    }
                }
            })
    }


    render() {
        return (
            <Formik innerRef={this.props.forwardedRef}
                    enableReinitialize
                    initialValues={this.state.initial}
                    validationSchema={yup.object({
                        firstName: yup
                            .string()
                            .required("پر کردن این فیلد اجباری است")
                            .max(30, "حداکثر 30 حرف میتوانید وارد کنید")
                            .min(1, "حداقل 1 حرف میتوانید وارد کنید")
                            .matches(/^[\u0600-\u06FF\s]+$/, "تنها استفاده از حروف فارسی مجاز است"),
                        lastName: yup
                            .string()
                            .required("پر کردن این فیلد اجباری است")
                            .max(30, "حداکثر 30 حرف میتوانید وارد کنید")
                            .min(1, "حداقل 1 حرف میتوانید وارد کنید")
                            .matches(/^[\u0600-\u06FF\s]+$/, "تنها استفاده از حروف فارسی مجاز است"),
                        username: yup
                            .string()
                            .required("پر کردن این فیلد اجباری است")
                            .max(30, "حداکثر 30 حرف میتوانید وارد کنید")
                            .min(1, "حداقل 1 حرف میتوانید وارد کنید")
                            .matches(/^[a-zA-Z0-9-_]+$/, "تنها استفاده از حروف انگلیسی مجاز است"),
                        phoneNumber: yup
                            .string()
                            .nullable(true)
                            .matches(/^[+]*[(]{0,1}[0-9]{1,3}[)]{0,1}[-\s\./0-9]*$/g, "شماره ی معتبر وارد کنید"),
                    })}
                    onSubmit={(values, actions) => {
                            this.props.postData(values)
                            this.postData(values)
                            actions.setSubmitting(false);
                    }}
            >
                {props => (
                    <form className={Style.formContainer} onSubmit={props.handleSubmit}>
                        <Grid className="animate__animated animate__fadeIn" container>
                            {this.items(props)}
                        </Grid>
                    </form>
                )}
            </Formik>

        );
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(PrivateInfo)
