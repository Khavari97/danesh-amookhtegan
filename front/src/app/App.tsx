import React from "react";
import "./App.scss";
import AppRouter from "./navigation/AppRouter";
import {SnackbarProvider} from 'notistack';
import {SnackbarUtils} from "./components/material-ui/Snackbar/SnackbarUtils";
import ThemeProvider from "./assets/styles/material/material";

export interface AppProps {
}

export interface AppState {
}

class App extends React.Component<AppProps, AppState> {
    constructor(props: AppProps) {
        super(props);
        this.state = {};
    }

    componentDidMount = () => {
    }

    render() {
        return (
            <SnackbarProvider maxSnack={3} anchorOrigin={{horizontal: 'right', vertical: 'bottom'}}>
                <SnackbarUtils/>
                <ThemeProvider>
                    <AppRouter/>
                </ThemeProvider>
            </SnackbarProvider>
        );
    }
}

export default App;
