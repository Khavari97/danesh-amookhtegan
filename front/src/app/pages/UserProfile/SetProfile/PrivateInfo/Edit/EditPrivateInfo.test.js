import React from 'react';
import App from './../../SetProfile';
import EditPrivateInfo from './EditPrivateInfo'
import renderer from "react-test-renderer";
import { BrowserRouter } from 'react-router-dom'
import Enzyme, { shallow, configure, mount } from "enzyme";
import { Provider } from "react-redux";
import { render, screen, prettyDOM, act, waitFor, wait, fireEvent, queryByAttribute } from '@testing-library/react'
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

const middlewares = [thunk]
const mockStore = configureStore(middlewares);
configure({ adapter: new Adapter() });

let mock_data = {
    province: [{ code: "7", id: 7, name: "بوشهر" }, { code: "8", id: 8, name: "تهران" }],
    data: [{ firstName: "آتنا", lastName: "گنجی", username: "atena", email: "atena.ganji.ag@gmail.com", province: "تهران", university: "خوارزمی", city: "تهران", cityObject: 1924, provinceObject: 8, universityObject: 1, phoneNumber: "+989011625458", genderDisplay: "زن", gender: "FEMALE" }],
    description: "دانشجوی دانشگاه خوارزمی ترم 6"
}

const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-testid="${val}"]`);
};

describe('render correctly page', () => {

    let mock_store = mockStore({
        userProfile: mock_data,
    });

    mock_store.dispatch = jest.fn()
    let wrapper = mount(<BrowserRouter>
        <Provider store={mock_store}>
            <EditPrivateInfo />
        </Provider>
    </BrowserRouter>)

    it('check render of page ', () => {
        expect(wrapper.find(EditPrivateInfo).exists()).toBe(true);
        expect(wrapper.find(EditPrivateInfo).length).toBe(1);
    });
    it('check render textfileds of page', () => {
        expect(findByTestAttr(wrapper, "firstName").exists()).toBe(true);
        expect(findByTestAttr(wrapper, "lastName").exists()).toBe(true);
        expect(findByTestAttr(wrapper, "gender").exists()).toBe(true);
        expect(findByTestAttr(wrapper, "username").exists()).toBe(true);
        expect(findByTestAttr(wrapper, "email").exists()).toBe(true);
        expect(findByTestAttr(wrapper, "province").exists()).toBe(true);
        expect(findByTestAttr(wrapper, "city").exists()).toBe(true);
        expect(findByTestAttr(wrapper, "university").exists()).toBe(true);
        expect(findByTestAttr(wrapper, "phoneNumber").exists()).toBe(true);
    });
    test('snapshot', () => {
        const snapshot = renderer.create(wrapper).toJSON();
        expect(snapshot).toMatchSnapshot();
    })
});




describe('change textarea component by render set profile to show edit private info', () => {
    let mock_store

    beforeEach(() => {
        mock_store = mockStore({
            userProfile: mock_data
        });
    });


    test('on change text fields by user event', async () => {
        const getById = queryByAttribute.bind(null, 'id');
        const handleSubmit = jest.fn();

        const dom = render(
            <BrowserRouter>
                <Provider store={mock_store}>
                    <App postData={handleSubmit} />
                </Provider>
            </BrowserRouter>
        );

        act(() => {
            fireEvent.click(getById(dom.container, 'infoEditButton'))
        })

        act(() => {
            fireEvent.change(screen.getByTestId('firstName').querySelector('input'), { target: { value: 'تستیییی' } });
        })
        expect(screen.getByTestId('firstName').querySelector('input').value).not.toBe('');
        expect(screen.getByTestId('firstName').querySelector('input').value).toBe('تستیییی');

        act(() => {
            fireEvent.change(screen.getByTestId('lastName').querySelector('input'), { target: { value: 'تستیییی' } });
        })
        expect(screen.getByTestId('lastName').querySelector('input').value).not.toBe('');
        expect(screen.getByTestId('lastName').querySelector('input').value).toBe('تستیییی');

        act(() => {
            fireEvent.change(screen.getByTestId('username').querySelector('input'), { target: { value: 'admin' } });
        })
        expect(screen.getByTestId('username').querySelector('input').value).not.toBe('');
        expect(screen.getByTestId('username').querySelector('input').value).toBe('admin');

        act(() => {
            fireEvent.change(screen.getByTestId('phoneNumber').querySelector('input'), { target: { value: '09011625458' } });
        })
        expect(screen.getByTestId('phoneNumber').querySelector('input').value).not.toBe('');
        expect(screen.getByTestId('phoneNumber').querySelector('input').value).toBe('09011625458');

        act(() => {
            fireEvent.click(getById(dom.container, 'infoEditButton'))
        })
    })


})