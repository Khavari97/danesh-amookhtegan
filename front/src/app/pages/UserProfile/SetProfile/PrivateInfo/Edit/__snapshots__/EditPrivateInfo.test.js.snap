// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly page snapshot 1`] = `
<form
  className="formContainer"
  onSubmit={[Function]}
>
  <div
    className="MuiGrid-root animate__animated animate__fadeIn MuiGrid-container"
  >
    <div
      className="MuiGrid-root gridItems MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        className="MuiGrid-root gridInputs MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
      >
        <div
          className="MuiFormControl-root MuiTextField-root Input-root-1"
          data-test="firstName"
          data-testid="firstName"
          onMouseOver={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
          >
            نام
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="firstName"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  نام
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root gridInputs MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
      >
        <div
          className="MuiFormControl-root MuiTextField-root Input-root-1"
          data-test="lastName"
          data-testid="lastName"
          onMouseOver={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
          >
            نام خانوادگی
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="lastName"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  نام خانوادگی
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root gridItems MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        className="MuiGrid-root gridInputs MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
      >
        <div
          className="MuiFormControl-root makeStyles-formControl-6"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            id="demo-simple-select-outlined-label"
          >
            جنسیت
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            data-testid="select"
            onClick={[Function]}
            onSelect={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-outlined-label mui-component-select-age"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              id="mui-component-select-age"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              data-testid="gender"
              id="demo-controlled-open-select"
              name="age"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value=""
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  جنسیت
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root gridItems MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        className="MuiGrid-root gridInputs MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
      >
        <div
          className="MuiFormControl-root MuiTextField-root Input-root-1"
          data-test="username"
          data-testid="username"
          onMouseOver={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
          >
            نام کاربری
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="username"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  نام کاربری
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root gridItems MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        data-testid="email"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
        >
          <p
            className="title"
          >
            ایمیل
          </p>
        </li>
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
        >
          <p
            className="description"
          />
        </li>
      </div>
      <div>
        <button
          aria-label="delete"
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <img
              alt=""
              src="alert.svg"
              style={
                Object {
                  "width": "20px",
                }
              }
            />
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiGrid-root gridItems MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        className="MuiGrid-root gridInputs MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
      >
        <div
          className="MuiFormControl-root MuiTextField-root Input-root-1"
          data-test="province"
          data-testid="province"
          onMouseOver={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
          >
            استان
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="province"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  استان
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root gridInputs MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
      >
        <div
          className="MuiFormControl-root MuiTextField-root Input-root-1"
          data-test="city"
          data-testid="city"
          onMouseOver={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
          >
            شهر
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="city"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  شهر
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root gridItems MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        className="MuiGrid-root gridInputs MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
      >
        <div
          className="MuiFormControl-root MuiTextField-root Input-root-1"
          data-test="university"
          data-testid="university"
          onMouseOver={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
          >
            دانشگاه
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="university"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  دانشگاه
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root gridItems MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        className="MuiGrid-root gridInputs MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
      >
        <div
          className="MuiFormControl-root MuiTextField-root Input-root-1"
          data-test="phoneNumber"
          data-testid="phoneNumber"
          onMouseOver={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink={true}
          >
            شماره تلفن
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="phoneNumber"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value="+98"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4 PrivateNotchedOutline-legendNotched-5"
              >
                <span>
                  شماره تلفن
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;
